sc-memory/sc-core/CMakeLists.txt

```
@@ -8,5 +8,8 @@ target_include_directories(sc-core
   PRIVATE ${GLIB2_INCLUDE_DIRS} 
   PRIVATE ${RocksDB_INCLUDE_DIR}
   )
-target_link_libraries(sc-core ${GLIB2_LIBRARIES} ${RocksDB_LIBRARIES} gcov)
+target_link_libraries(sc-core ${GLIB2_LIBRARIES} ${RocksDB_LIBRARIES})
+if(${SC_COVERAGE} AND CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+	target_link_libraries(sc-core gcov)
```

This change would allow us to fail properly when COVERAGE + `clang` is used.
```suggestion
if(${SC_COVERAGE})
  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
	  target_link_libraries(sc-core gcov)
  else()
    message(SEND_ERROR "Currently, code coverage is supported only with GCC as a compiler")
  endif()
endif()
```